src/core/default_appearance.js

```diff
@@ -87,6 +87,94 @@ function parseDefaultAppearance(str) {
   return new DefaultAppearanceEvaluator(str).parse();
 }
 
+class AppearanceEvaluator extends EvaluatorPreprocessor {
+  constructor(str) {
+    super(str);
+    this.str = str;
+  }
+
+  parse() {
+    const operation = {
+      fn: 0,
+      args: [],
+    };
+    const result = {
+      fontSize: 0,
+      fontName: "",
+      fontColor: /* black = */ new Uint8ClampedArray(3),
+    };
+
+    let isInTextSection = false;
+    let scaleFactor = 1;
+
+    try {
+      while (true) {
+        operation.args.length = 0; // Ensure that `args` it's always reset.
+
+        if (!this.read(operation)) {
+          break;
+        }
+        const { fn, args } = operation;
+
+        switch (fn | 0) {
+          case OPS.beginText:
+            isInTextSection = true;
+            break;
+          case OPS.endText:
+            isInTextSection = false;
+            scaleFactor = 1;
+            break;
+          case OPS.setTextMatrix:
+            scaleFactor *= Math.abs(args[0]);
+            break;
+          case OPS.setFont:
+            if (!isInTextSection) {
+              break;
+            }
+            const [fontName, fontSize] = args;
+            if (fontName instanceof Name) {
+              result.fontName = fontName.name;
+            }
+            if (typeof fontSize === "number" && fontSize > 0) {
+              result.fontSize = fontSize * scaleFactor;
+            }
+            break;
+          case OPS.setFillRGBColor:
+            if (!isInTextSection) {
+              break;
+            }
+            ColorSpace.singletons.rgb.getRgbItem(args, 0, result.fontColor, 0);
+            break;
+          case OPS.setFillGray:
+            if (!isInTextSection) {
+              break;
+            }
+            ColorSpace.singletons.gray.getRgbItem(args, 0, result.fontColor, 0);
+            break;
+          case OPS.setFillColorSpace:
+            if (!isInTextSection) {
+              break;
+            }
+            ColorSpace.singletons.cmyk.getRgbItem(args, 0, result.fontColor, 0);
+            break;
+        }
+      }
+    } catch (reason) {
+      warn(`parseDefaultAppearance - ignoring errors: "${reason}".`);
```

Comment:
Too much copy-paste here:
```suggestion
      warn(`parseAppearanceStream - ignoring errors: "${reason}".`);
```